snapcraft/extensions/kde_neon_6.py

```diff
@@ -0,0 +1,272 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2022 Canonical Ltd.
+#           2023 Scarlett Moore <sgmoore@kde.org>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Generic KDE NEON extension to support core22 and onwards."""
+import dataclasses
+import functools
+import re
+from typing import Any, Dict, List, Optional, Tuple
+
+from overrides import overrides
+
+from .extension import Extension, get_extensions_data_dir, prepend_to_env
+
+_QT6_SDK_SNAP = {"core22": "kde-qt6-core22-sdk"}
+_KF6_SDK_SNAP = {"core22": "kf6-core22-sdk"}
```

Comment:
I gather we're safe on backwards compatibility of the ABIs for both? (No need for minor version specifications?)

Reply:
This is fine. Previously we were using packages and had to ensure the package versions were aligned. We are now building all the parts from source. If there are incompatible versions it is a snap problem that needs attention.

Reply:
Awesome, thanks!

Reply:
this is great news!